contracts/cw-core/src/tests.rs

```diff
@@ -2230,3 +2266,107 @@ fn test_execute_stargate_msg() {
     assert!(res.is_err());
 }
+
+#[test]
+fn test_module_prefixes() {
+    let mut app = App::default();
+    let govmod_id = app.store_code(sudo_proposal_contract());
+    let gov_id = app.store_code(cw_core_contract());
+
+    let govmod_instantiate = cw_proposal_sudo::msg::InstantiateMsg {
+        root: CREATOR_ADDR.to_string(),
+    };
+
+    let gov_instantiate = InstantiateMsg {
+        admin: None,
+        name: "DAO DAO".to_string(),
+        description: "A DAO that builds DAOs.".to_string(),
+        image_url: None,
+        automatically_add_cw20s: true,
+        automatically_add_cw721s: true,
+        voting_module_instantiate_info: ModuleInstantiateInfo {
+            code_id: govmod_id,
+            msg: to_binary(&govmod_instantiate).unwrap(),
+            admin: Admin::CoreContract {},
+            label: "voting module".to_string(),
+        },
+        proposal_modules_instantiate_info: vec![
+            ModuleInstantiateInfo {
+                code_id: govmod_id,
+                msg: to_binary(&govmod_instantiate).unwrap(),
+                admin: Admin::CoreContract {},
+                label: "proposal module 1".to_string(),
+            },
+            ModuleInstantiateInfo {
+                code_id: govmod_id,
+                msg: to_binary(&govmod_instantiate).unwrap(),
+                admin: Admin::CoreContract {},
+                label: "proposal module 2".to_string(),
+            },
+            ModuleInstantiateInfo {
+                code_id: govmod_id,
+                msg: to_binary(&govmod_instantiate).unwrap(),
+                admin: Admin::CoreContract {},
+                label: "proposal module 2".to_string(),
+            },
+        ],
+        initial_items: None,
+    };
+
+    let gov_addr = app
+        .instantiate_contract(
+            gov_id,
+            Addr::unchecked(CREATOR_ADDR),
+            &gov_instantiate,
+            &[],
+            "cw-governance",
+            None,
+        )
+        .unwrap();
+
+    let modules: Vec<ProposalModule> = app
+        .wrap()
+        .query_wasm_smart(
+            gov_addr,
+            &QueryMsg::ProposalModules {
+                start_after: None,
+                limit: None,
+            },
+        )
+        .unwrap();
+
+    assert_eq!(modules.len(), 3);
+
+    let module_1 = &modules[0];
+    assert_eq!(module_1.status, ProposalModuleStatus::Active {});
+    assert_eq!(module_1.prefix, "A");
+    assert_eq!(&module_1.address, &modules[0].address);
+
+    let module_2 = &modules[1];
+    assert_eq!(module_2.status, ProposalModuleStatus::Active {});
+    assert_eq!(module_2.prefix, "B");
+    assert_eq!(&module_2.address, &modules[1].address);
+
+    let module_3 = &modules[2];
+    assert_eq!(module_3.status, ProposalModuleStatus::Active {});
+    assert_eq!(module_3.prefix, "C");
+    assert_eq!(&module_3.address, &modules[2].address);
+}
+
+fn get_active_modules(app: &App, gov_addr: Addr) -> Vec<ProposalModule> {
```

Comment:
How would you feel about having a query for this? This is likely what frontends will care about most.

Reply:
sure thing